### PR TITLE
Refactor Named Tuples and API Documentation

### PR DESCRIPTION
**Pull request by hotzenklotz**
More house keeping:
- use standard named tuples and class for all objects
- renamed all internal methods
- added a few more type hints
- add doc strings
- generate API Docs automagically with `pydoc-markdown`
- bumped package version to `1.0.0` since the whole refactoring might be a breaking change and looks more serious :-D

Fixes https://github.com/scalableminds/wknml/issues/33, #28

### CODE REVIEW

**Comment by hotzenklotz** [comment]
@jstriebel I am pretty sure that the changes to classes/named tuples shouldn't break anything, but I would like a second opinion.

File: wknml/__init__.py [hunk 1]
```diff
@@ -1,106 +1,161 @@
...
+
+    Attributes:
+        id: int
+        name: str
+        children: List[Group]
+    """
+
+    id: int
+    name: str
+    children: List["Group"]
```
Comment (normanrz) [inline wknml/__init__.py:123]: Can't self-reference a type?

**Review by normanrz — COMMENTED** [review]
I am not sure if any of the `parse_*` or `dump_*` functions have been used anywhere else? Maybe you can to a quick grep on voxelytics?

File: wknml/__init__.py (hunk 1 above)
Comment (hotzenklotz) [reply wknml/__init__.py:123]: ne, weil der an der Stelle gerade erst definiert wird. mypy versteht das

**Comment by hotzenklotz** [comment]
> I am not sure if any of the `parse_*` or `dump_*` functions have been used anywhere else? Maybe you can to a quick grep on voxelytics?

I did a quicky search and didn't see any use of those methods.

**Review by normanrz — APPROVED** [review]

**Review by normanrz — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -57,6 +66,11 @@ PyPi releases are automatically pushed when creating a new Git tag/Github releas
 poetry version <patch, minor, major>
 ```
 
+If necessary, rebuild the documentation and commit to repository:
+```
+poetry run pydoc-markdown -m wknml --render-toc > docs/wknml.md
```
Comment (jstriebel) [inline README.md:71]: Please add a check in the CI that this is up-to-date :white_check_mark:

**Review by jstriebel — COMMENTED** [review]
Nice! Please just add a CI check for the docs, so one is forced to update them if necessary. Besides this, LGTM :+1:

File: .github/workflows/python-module.yml [hunk 3]
```diff
@@ -18,10 +18,14 @@ jobs:
...
         poetry run black --check wknml tests examples
-    - name: Test
+    - name: Unit Tests
       run: |
         mkdir -p testoutput
         poetry run pytest tests
+    - name: Check Documentation for updates
+      run: |
+        poetry run pydoc-markdown -m wknml --render-toc > docs/ci_test.md
+        diff docs/ci_test.md docs/wknml.md
```
Comment (jstriebel) [inline .github/workflows/python-module.yml:28]:
```suggestion
        diff docs/ci_test.md docs/wknml.md
        rm docs/ci_test.md
```

**Review by jstriebel — APPROVED** [review]
Nice! I'm also quite sure that this does not make a difference in any of our use-cases, only some internals are probably changed.

See my one suggestion above, everything else LGTM :+1:

**Pull request merged**